### PR TITLE
[Metricbeat] Add PubSub metricset to Google Cloud Platform module

### PR DESCRIPTION
**Pull request by sayden**
Adds PubSub metricset to Google Cloud Platform module.

Requires a slight change in code to adapt the Stackdriver filter which doesn't allow to specify a region when requesting metrics from PubSub service.

Everything else is docs and yaml

cf. elastic/beats#15812

### CODE REVIEW

File: x-pack/metricbeat/module/googlecloud/pubsub/_meta/data_subscription.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+    "@timestamp": "2020-01-14T11:46:00.000Z",
+    "@metadata": {
+        "beat": "metricbeat",
+        "type": "_doc",
+        "version": "8.0.0"
+    },
+    "googlecloud": {
+        "labels": {
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/googlecloud/pubsub/_meta/data_subscription.json]: just curious is the label key here resource.subscription_id? I understand the value is `test-subscription`, but not sure whats the `resource` part. Thanks!

File: x-pack/metricbeat/module/googlecloud/pubsub/_meta/fields.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        - name: pull_ack_message_operation_count
+          type: long
+          description: Cumulative count of acknowledge message operations, grouped by result. For a definition of message operations, see Cloud Pub/Sub metric subscription/mod_ack_deadline_message_operation_count.
+        - name: pull_ack_request_count
+          type: long
+          description: Cumulative count of acknowledge requests, grouped by result.
+        - name: pull_message_operation_count
+          type: long
+          description: Cumulative count of pull message operations, grouped by result. For a definition of message operations, see Cloud Pub/Sub metric subscription/mod_ack_deadline_message_operation_count.
+        - name: pull_request_count
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/googlecloud/pubsub/_meta/fields.yml]: do you think merging some of these fields will be better? For example, pull_request_count and push_request_count can be request_count.pull and request_count.push.

File: x-pack/metricbeat/module/googlecloud/pubsub/_meta/data_subscription.json (hunk 1 above)
Comment (sayden) [reply x-pack/metricbeat/module/googlecloud/pubsub/_meta/data_subscription.json]: This is more like a property but we thought it was a bit too "complex" to talk about metadata, labels, resources, properties... so we joined all of them into "labels".

This information comes on each [response](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries) of the Stackdriver API. The one you mention here is a resource label as you can see in the link (follow [MonitoredResource](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.uptimeCheckConfigs#MonitoredResource) link too). You can also find "system" labels, "user" labels and instead of having a bunch of `user.labels.*`, `resource.labels.*`, `system.labels.*` we swapped the `.labels` part to just have everything under the key `labels.*`

For example, resource labels in `compute` are the instance type or the storage type. User labels are a key-value you can set in gcp console.

In this case, PubSub don't have the specific implementation to gather `user` labels so we are missing the data that the user might input in the console. This data uses PubSub API and not Stackdriver API, that's why each metricset requires an specific implementation inside the stackdriver metricset

I'm not sure if I did a very good job explaining this now, frankly :sweat_smile:

File: x-pack/metricbeat/module/googlecloud/pubsub/_meta/fields.yml (hunk 2 above)
Comment (sayden) [reply x-pack/metricbeat/module/googlecloud/pubsub/_meta/fields.yml]: Definitely but, because we are using lightweight modules, no field naming modification "is possible" (or, at least, not without making code way more complex now)

File: metricbeat/docs/modules/googlecloud.asciidoc [hunk 3]
```diff
@@ -83,6 +83,7 @@ metricbeat.modules:
 - module: googlecloud
   metricsets:
     - compute
```
Comment (kaiyan-sheng) [inline metricbeat/docs/modules/googlecloud.asciidoc]: loadbalancing seems to be missing in the default config here?

File: x-pack/metricbeat/metricbeat.reference.yml [hunk 4]
```diff
@@ -411,6 +411,7 @@ metricbeat.modules:
 - module: googlecloud
   metricsets:
     - compute
+    - pubsub
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/metricbeat.reference.yml]: loadbalancing is missing here?

File: x-pack/metricbeat/module/googlecloud/stackdriver/metrics_requester.go [hunk 5]
```diff
@@ -52,11 +52,13 @@ type stackdriverMetricsRequester struct {
...
 	out = make([]*monitoringpb.TimeSeries, 0)
 
+	filter := r.getFilterForMetric(m)
+
 	req := &monitoringpb.ListTimeSeriesRequest{
 		Name:     "projects/" + r.config.ProjectID,
 		Interval: r.interval,
 		View:     monitoringpb.ListTimeSeriesRequest_FULL,
-		Filter:   r.getFilterForMetric(m),
+		Filter:   filter,
```
Comment (kaiyan-sheng) [inline x-pack/metricbeat/module/googlecloud/stackdriver/metrics_requester.go]: nit: is this change necessary? 😬

Comment (sayden) [reply x-pack/metricbeat/module/googlecloud/stackdriver/metrics_requester.go]: I might mess it up solving the conflicts. Glad that you spot it to take a closer look :grimacing:

File: metricbeat/docs/modules/googlecloud.asciidoc (hunk 3 above)
Comment (sayden) [reply metricbeat/docs/modules/googlecloud.asciidoc]: Added to default, thanks!

File: x-pack/metricbeat/metricbeat.reference.yml (hunk 4 above)
Comment (sayden) [reply x-pack/metricbeat/metricbeat.reference.yml]: Yeap! Added!

File: x-pack/metricbeat/module/googlecloud/stackdriver/metrics_requester.go (hunk 5 above)
Comment (sayden) [reply x-pack/metricbeat/module/googlecloud/stackdriver/metrics_requester.go]: I think it was only that, I reverted it :grimacing:
